File: agent-driven-architecture-main/examples/typescript/src/blocks/escalation/adapter/adapter.ts
// ── blocks/escalation/adapter/adapter — THIS BLOCK'S IMPURE UNIT (G13) ────
// Anything holding a client, a DB handle or a socket lives here — one clearly
// named place per block, never inline in a tool. The DB call ships INSIDE the
// block as port+adapter (4.6/G11); `app/wire` is the only file that decides
// which adapter is real.
//
// The "client" below is a console pager, because this reference runs offline.
// In a real deployment it is a PagerDuty SDK, and nothing else about the block
// changes.

import type { Emit } from "@adr/spine/pure/emit";
import type { TicketId } from "@adr/spine/pure/ids";
import type { OncallPort } from "../port";

export function livePager(emit: Emit): OncallPort {
  return {
    page: (ticket: TicketId) => emit(`[pager] on-call paged for ticket ${ticket}`),
  };
}

File: agent-driven-architecture-main/examples/kotlin/block/escalation/adapter/src/main/kotlin/adr/blocks/escalation/Adapter.kt
// ── blocks/escalation/adapter — the ONLY file in this block that may hold a client ─
// G13 in one file name: if you want to know what this block can touch in the world,
// this is the only place to look. It is bound at the single composition root (G7).

package adr.blocks.escalation

import adr.spine.pure.Emit
import adr.spine.pure.TicketId

/**
 * The rim of the block. `send` stands in for the pager SDK — a real one would be a
 * client handle held here and nowhere else.
 */
class LivePager(
    private val endpoint: String,
    private val send: Emit<String>,
) : OncallPort {
    override fun page(ticket: TicketId) {
        send("POST $endpoint {\"ticket\":\"${ticket.value}\"}")
    }
}
